src/stan/math/prim/mat/meta/index_type.hpp

```diff
@@ -21,6 +21,16 @@ namespace stan {
       typedef typename Eigen::Matrix<T,R,C>::Index type;
     };
 
+    template <typename T>
+    struct index_type<Eigen::MatrixBase<T> > {
+      typedef typename index_type<T>::type type;
+    };
+
+    template <typename T>
+    struct index_type<Eigen::Map<T> > {
```

Comment:
Sorry, I get confused with Eigen doc. I still haven't found how to
specialize matrix multiplication.

The question I have is... is there a single templated function we can write
to cover both Eigen::MatrixBase and Eigen::Map?


On Wed, Mar 11, 2015 at 8:09 PM, Bob Carpenter <notifications@github.com>
wrote:

> In src/stan/math/prim/mat/meta/index_type.hpp
> <https://github.com/stan-dev/stan/pull/1210#discussion_r26268589>:
>
> > @@ -21,6 +21,16 @@ namespace stan {
> >        typedef typename Eigen::Matrix<T,R,C>::Index type;
> >      };
> >
> > +    template <typename T>
> > +    struct index_type<Eigen::MatrixBase<T> > {
> > +      typedef typename index_type<T>::type type;
> > +    };
> > +
> > +    template <typename T>
> > +    struct index_type<Eigen::Map<T> > {
>
> Yes. I'd suggest looking these things up! Eigen has great doc. Map<T> is
> the expression template class that maps contiguous memory to an Eigen type
> T. - Bob
> … <#14c0b507620a87f6_>
>  On Mar 12, 2015, at 7:24 AM, Daniel Lee <notifications@github.com>
> wrote: In src/stan/math/prim/mat/meta/index_type.hpp: > @@ -21,6 +21,16 @@
> namespace stan { > typedef typename Eigen::Matrix<T,R,C>::Index type; > };
> > > + template <typename T> > + struct index_type<Eigen::MatrixBase<T> > {
> > + typedef typename index_type<T>::type type; > + }; > + > + template
> <typename T> > + struct index_type<Eigen::Map<T> > { Is Eigen::Map<T>
> different than Eigen::MatrixBase<T>? — Reply to this email directly or view
> it on GitHub.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/stan-dev/stan/pull/1210/files#r26268589>.
>